src/Pim/Bundle/ReferenceDataBundle/Elasticsearch/Filter/Attribute/ReferenceDataFilter.php

```diff
@@ -0,0 +1,178 @@
+            $this->checkValue($attribute, $values);
+        }
+
+        $attributePath = $this->getAttributePath($attribute, $locale, $channel);
+
+        switch ($operator) {
+            case Operators::IN_LIST:
+                $clause = [
+                    'terms' => [
+                        $attributePath => $values,
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::IS_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($clause);
+                break;
+
+            case Operators::IS_NOT_EMPTY:
+                $clause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addFilter($clause);
+                break;
+
+            case Operators::NOT_IN_LIST:
+                $mustNotClause = [
+                    'terms' => [
+                        $attributePath => $values,
+                    ],
+                ];
+                $filterClause = [
+                    'exists' => [
+                        'field' => $attributePath,
+                    ],
+                ];
+                $this->searchQueryBuilder->addMustNot($mustNotClause);
+                $this->searchQueryBuilder->addFilter($filterClause);
+                break;
+
+            default:
+                throw InvalidOperatorException::notSupported($operator, static::class);
+        }
+
+        return $this;
+    }
+
+    /**
+     * @param AttributeInterface $attribute
+     *
+     * @return bool
+     */
+    public function supportsAttribute(AttributeInterface $attribute)
+    {
+        $referenceDataName = $attribute->getReferenceDataName();
+
+        $isRegistredReferenceData = null !== $referenceDataName &&
+            !empty($referenceDataName) &&
+            null !== $this->registry->get($attribute->getReferenceDataName());
+
+        return $isRegistredReferenceData;
+    }
+
+    /**
+     * Check if values are valid
+     *
+     * @param AttributeInterface $attribute
+     * @param mixed              $values
+     *
+     * @throws ObjectNotFoundException
+     */
+    protected function checkValue(AttributeInterface $attribute, $values)
+    {
+        FieldFilterHelper::checkArray($attribute->getCode(), $values, static::class);
+
+        foreach ($values as $value) {
+            FieldFilterHelper::checkIdentifier($attribute->getCode(), $value, static::class);
+        }
+
+        $referenceDataRepository = $this->referenceDataRepositoryResolver->resolve($attribute->getReferenceDataName());
+        $existingReferenceData =  $referenceDataRepository->findCodesByIdentifiers($values);
+        $referenceDataCodes = array_map(
```

Comment:
useful because findCodesByIdentifiers return an array like that `[0 => ['code' => 'code1'], 1  => ['code' => 'code2']]`